src/transformers/data/data_collator.py

```diff
@@ -34,6 +34,22 @@
 """
 DataCollator = NewType("DataCollator", Callable[[List[InputDataClass]], Dict[str, Any]])
 
+def pad_no_warning(tokenizer: PreTrainedTokenizerBase, *pad_args, **pad_kwargs):
+    """ 
+    Pad without warning the user.
+    """
+    
+    # Save the state of the warning, then disable it
+    warning_state = tokenizer.deprecation_warnings.get("Asking-to-pad-a-fast-tokenizer", False)
+    tokenizer.deprecation_warnings["Asking-to-pad-a-fast-tokenizer"] = True
+    
+    padded = tokenizer.pad(*pad_args, **pad_kwargs)
+    
+    # Restore the state of the warning.
+    tokenizer.deprecation_warnings["Asking-to-pad-a-fast-tokenizer"] = warning_state
```

Comment:
This should probbly be in a `try`/`finally` block so that the deprecation warning is put back even in case of error.